hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -441,9 +441,10 @@ public void refreshTimeline() throws IOException {
       return null;
     }
 
+    jssc.setJobGroup(this.getClass().getSimpleName(), "Checking if input is empty");
     if ((!avroRDDOptional.isPresent()) || (avroRDDOptional.get().isEmpty())) {
       LOG.info("No new data, perform empty commit.");
-      return Pair.of(schemaProvider, Pair.of(checkpointStr, jssc.emptyRDD()));
+      return Pair.of(schemaProvider, Pair.of(checkpointStr, null));
```

Comment:
why do we need modify this?

Reply:
It saves one call to isEmpty() in `DeltaSync::writeToSink`, which could further eliminate one spark job.

Reply:
> It saves one call to isEmpty() in `DeltaSync::writeToSink`, which could further eliminate one spark job.

You would separate it in another PR.

Reply:
got it, will move it to a new PR.